modules/services/polybar.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.polybar;
+
+in
+
+{
+  options = {
+    services.polybar = {
+      enable = mkEnableOption "Polybar status bar";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.polybar;
+        defaultText = "pkgs.polybar";
+        description = "Polybar package to install.";
+        example =  literalExample ''
+          pkgs.polybar.override {
+            i3GapsSupport = true;
+            alsaSupport = true;
+            iwSupport = true;
+            githubSupport = true;
+          }
+        '';
+      };
+
+      config = mkOption {
+        # type = types.either (types.attrsOf types.attrs) types.path;
+        type = types.coercedTo
+          types.path
+          (p: { "section/base" = { include-file = "${p}"; }; })
+          (types.attrsOf types.attrs);
+        description = ''
+          Polybar configuration. Can be either path to a file, or set of attibutes
+          that will be used to create the final configuration.
+        '';
+        default = {};
+        example = literalExample ''
+          "bar/top" = {
+            monitor = "\''${env:MONITOR:eDP1}";
+            width = "100%";
+            height = "3%";
+            radius = 0;
+            modules-center = "date";
+          };
+
+          "module/date" = {
+            type = "internal/date";
+            internal = 5;
+            date = "%d.%m.%y";
+            time = "%H:%M";
+            label = "%time%  %date%";
+          };
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        description = "Additional configuration to add.";
+        default = "";
+        example = literalExample ''
+          [module/date]
+          type = internal/date
+          interval = 5
+          date = "%d.%m.%y"
+          time = %H:%M
+          format-prefix-foreground = \''${colors.foreground-alt}
+          label = %time%  %date%
+        '';
+      };
+
+      startScript = mkOption {
+        type = types.lines;
+        description = ''
+          This script will be used to start the polybars.
+          Set all necessary environment variables here and start all bars.
+          It can be assumed that 'polybar' executable is in the PATH.
```

Comment:
Can replace `'polybar'` by `<command>polybar</command>` and `PATH` by `<envar>PATH</envar>`.